server/src/main/java/com/testsigma/controller/api/agent/AgentDevicesController.java

```diff
@@ -130,16 +137,34 @@ public IosWdaResponseDTO deviceWdaUrl(@PathVariable String agentUuid, @PathVaria
     IosWdaResponseDTO iosWdaResponseDTO = new IosWdaResponseDTO();
     Agent agent = agentService.findByUniqueId(agentUuid);
     AgentDevice agentDevice = agentDeviceService.findAgentDeviceByUniqueId(agent.getId(), deviceUuid);
-    ProvisioningProfileDevice profileDevice = provisioningProfileDeviceService.findByAgentDeviceId(agentDevice.getId());
-    if (profileDevice != null) {
-      URL presignedUrl = storageServiceFactory.getStorageService().generatePreSignedURL("wda/"
-        + profileDevice.getProvisioningProfileId() + "/wda.ipa", StorageAccessLevel.READ, 180);
-      iosWdaResponseDTO.setWdaPresignedUrl(presignedUrl.toString());
-      log.info("Ios Wda Response DTO - " + iosWdaResponseDTO);
-      return iosWdaResponseDTO;
+    String presignedUrl;
+    if (!agentDevice.getIsEmulator()) {
```

Comment:
@tarun-testsigma move this to separate methods

Reply:
Moved to separate actions